### PR TITLE
Add webSocketsClient to different plugin components

### PR DESCRIPTION
**Pull request by mgdelacroix**
#### Summary
This PR adds the webSocketClient to different plugin components, so it can be used from plugins directly.

#### Release Note
```release-note
Added webSocketClient to Pluggable and PostWillRenderEmbed plugin registered components
```

### CODE REVIEW

**Review by jespino — APPROVED** [review]
LGTM

File: types/store/plugins.ts [hunk 1]
```diff
@@ -119,9 +121,10 @@ export type AdminConsolePluginComponent = {
 export type PostWillRenderEmbedPluginComponent = {
     id: string;
     pluginId: string;
-    component: React.ComponentType<{ embed: PostEmbed }>;
+    component: React.ComponentType<{ embed: PostEmbed; webSocketClient?: WebSocketClient }>;
     match: (arg: PostEmbed) => boolean;
     toggleable: boolean;
+
```
Comment (jespino) [inline types/store/plugins.ts:127]:
```suggestion
```
Unneeded extra line.

**Review by hmhealey — APPROVED** [review]
As discussed, passing the websocket client everywhere directly isn't ideal, but this is the lowest risk way of making this work for the time being. We'll hopefully replace this with some sort of context-based hook going forward, but that'll likely have it's own ramifications.

Also, we found some issues with how Focalboard is using the websocket and overwriting callbacks used by the web app to handle things like reconnects, but it's an existing issue that manages to work accidentally, so I'll be filing a separate ticket to fix that

**Edit:** Here's a ticket for that issue I mentioned on the web app side: https://mattermost.atlassian.net/browse/MM-45703

**Pull request merged**